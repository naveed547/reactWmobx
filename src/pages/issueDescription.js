import React, { Component } from 'react';
//import {bindActionCreators} from "redux";
//import {connect} from "react-redux";
//import * as actions from "../actions/repoAction";
import moment from 'moment';
import '../components/description.scss';
import { observer, inject } from 'mobx-react';

@inject("RepoStore","UserStore")

@observer
export class IssueDescription extends Component {
	constructor(props) {
		super(props);
		
		this.RepoStore = this.props.RepoStore;
		this.UserStore = this.props.UserStore;
	}
  	renderIssue = () => {
		let issue = this.RepoStore.repoData; 
		return (
			<div className="card">
				<div className="card-header row mx-0">
					<div className="col-sm-10 pl-0">#{issue.id} <b>{issue.title}</b></div>
					<div className={issue.state === 'open' ? 'col-sm-2 text-right text-danger pr-0':'col-sm-2 text-right text-success pr-0'}>
						<i className={issue.state === 'open' ? 'fa fa-exclamation-circle text-danger':'fa fa-check text-success'}></i>
						<span className="pl-1 text-capitalize">{issue.state}</span>
					</div>	
				</div>
				<div className="card-body">
					<p className="card-text">Issue Number: <b>{issue.number}</b></p>
					<p className="card-text">{issue.body}</p>
				</div>
				<div className="card-body row">
					<p className="card-text col-sm-6">
						<span>Created By: </span>
						<span><b>{issue.user.login}</b></span>
					</p>
					<p className="card-text col-sm-6 text-right">
						<small className="text-muted">Last updated <b>{moment(issue.state === 'open' ? issue.updated_at : issue.closed_at).fromNow()}</b>
						</small>
					</p>
				</div>
			</div>
		);
  }
  componentDidMount() {
	this.RepoStore.loadIssue(this.props.match.params.issueId,this.props.match.params.user,this.props.match.params.repo);
  }
  render() {
  	if (this.RepoStore 
  		&& this.RepoStore.repoData 
  		&& this.RepoStore.repoData.number) {
	    return (
	    	<div>
	      		<h2 className="my-5 text-center">Issue Description</h2>
	      		User Name: {this.RepoStore.userName} <br/>
	      		Repo Name: {this.RepoStore.currentRepo} <br/>
	      		{this.renderIssue()}
	      	</div>
	    );
	}else {
		return (
			<div className="my-5 text-center">Loading...</div>
		)
	}
  }
}


/* const mapStateToProps = state => {
    return {
        repoDetails: state.reposReducer
    }
};

const mapDispatchToProps = dispatch => {
    return bindActionCreators(actions, dispatch);
}


const IssueDescriptionContainer = connect(mapStateToProps, mapDispatchToProps)(IssueDescription);


export default IssueDescriptionContainer; */

