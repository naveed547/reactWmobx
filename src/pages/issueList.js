import React, { Component } from 'react';
//import {bindActionCreators} from "redux";
//import {connect} from "react-redux";
import '../components/list.scss';
//import {ISFMapper,TFMapper,LTMapper} from '../helper/stateActionMapper';
//import * as actions from "../actions/repoAction";
import InputSearchFilter from '../components/inputSearchFilter';
import TopFilter from '../components/topFilter';
import IssueTable from '../components/listTable';
import { observer, inject } from 'mobx-react';

@inject("RepoStore","UserStore")

@observer
export class IssueList extends Component {
	constructor(props) {
		super(props);
		
		this.RepoStore = this.props.RepoStore;
		this.UserStore = this.props.UserStore;
	}
	componentDidMount() {
		this.RepoStore.loadIssues(this.props.match.params.repo,this.props.match.params.user);
	}
	render() {
		if (this.RepoStore.repoData.total_count !== undefined) {
		    return (
				<React.Fragment>
					<h2 className="text-center my-5">Issues List <span className="my-5 text-center">{this.RepoStore.repoData.total_count ? `(${this.RepoStore.repoData.total_count})` : ''} </span></h2>
					User Name: <span className="user-name">{this.RepoStore.userName}</span> <br/>
	      			Repo Name: <span className="repo-name">{this.RepoStore.currentRepo}</span> <br/>
	      			{this.RepoStore.repoData.total_count ? (
				        <div className="card">
							<div className="card-header row mx-0">
								<div className="col-sm-4">
									<InputSearchFilter search={this.UserStore.searchIssues} />
								</div>
								<div className="col-sm-8">
									<TopFilter />
								</div>
							</div>
							<IssueTable />	
						</div>
				      ) : (
				        <div className="my-5 text-center">No data found for this user</div>
				      )}
				</React.Fragment>
		    );
		}
		else {
			return (
				<div className="my-5 text-center">Loading...</div>
			)
		}
	}

  
}