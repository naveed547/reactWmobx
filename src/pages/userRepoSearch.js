import React, { Component } from 'react';
import '../components/description.scss';
import { Link } from 'react-router-dom';
import { observer, inject } from 'mobx-react';

@inject("RepoStore")

@observer 
class UserRepoSearch extends Component {
  constructor(props) {
    super(props);
		this.userName = React.createRef();
		
		this.store = this.props.RepoStore;
  }
  selectRepo = event => {
  	event.preventDefault();
  	const user = this.userName.current.value;
    this.store.loadRepos(user);
  }
  componentDidMount() {
		this.store.loadRepos(this.store.userName);
  }
  renderForm = () => {
      return (
		<form name="repoSearch" className="form-inline justify-content-center" onSubmit={(e) => {e.preventDefault();this.store.loadRepos(this.userName.current.value)}}>
		  <div className="form-group mb-2 mr-2">
		    <label htmlFor="userName" className="sr-only">Password</label>
		    <input type="text" className="form-control" id="userName" onChange={() => {}} name="user" ref={this.userName} value={this.store.userName} placeholder="Search Git user name" />
		  </div>
		  <button type="submit" className="btn btn-primary mb-2">Search</button>
		</form>
      );
  }
  renderRepoList() {
  	if (!this.store.repoData.length) {
  		return (<div>Loading...</div>);
  	} else {
  		return this.store.repoData.map((repo,index) => {
  			return (
  				<li className="list-group-item" key={repo.id}>
						<Link to={`/issueList/${this.store.userName}/${repo.name}`}>
							{repo.name}
						</Link>
  				</li>
  			)
  		})
  	}
  }
  render() {
  	return (
			<div>
				<h2 className="my-5 text-center">Search User and Repo</h2>
				User Name: <span className="user-name">{this.store.userName}</span> <br/>
				<div className="card">
					<div className="card-header">
						 {this.renderForm()} 
					</div>
					<ul className="list-group list-group-flush repo-list">{this.renderRepoList()}</ul>
				</div>
			</div>
	    );
  } 
}
export default UserRepoSearch;

