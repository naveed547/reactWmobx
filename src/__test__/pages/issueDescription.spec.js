import React from "react";
//import {mount} from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import {IssueDescription} from "../../pages/issueDescription";
import initialState from '../../reducers/initialState';
import repoMock from "../data/mock";

describe("IssueDescription", () => {
    it('IssueDescription component renders correctly', () => {
        const props = {
            loadIssue: jest.fn(),
            repoDetails: {
                ...initialState.repoDetails,
                repoData:   repoMock.singleIssue
            },
            match: {
                params:{
                    issueId: repoMock.issueNumber,
                    user: repoMock.userName,
                    repo: repoMock.repoName
                }
            }
        };
        const rendered = renderer.create(
            <MemoryRouter>
                <IssueDescription {...props} />
            </MemoryRouter>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
    /* it("displays the userRepoSearch component when it has fetched users", () => {
        const props = {
            loadRepos: jest.fn(),
            repoDetails: {
                ...initialState.repoDetails,
                repoData:   repoMock.userRepos
            }
        };

        const container = mount(
            <MemoryRouter>
                <UserRepoSearch {...props} />
            </MemoryRouter>
        );
        expect(container.find('.user-name').text()).toEqual(props.repoDetails.userName);
        expect(container.find('.repo-list li').length).toEqual(props.repoDetails.repoData.length);
    });
  
    it("does not display the userRepoSearch when ther are no users", () => {
        const props = {
            loadRepos: jest.fn(),
            repoDetails: {
                userName: ' ',
		        currentRepo: ' ',
                repoData:   []
            }
        };
    
        const container = mount(
            <MemoryRouter>
                <UserRepoSearch {...props} />
            </MemoryRouter>
        );
        expect(container.find('.user-name').text()).toEqual(' ');
        expect(container.find('.repo-list li').length).toEqual(0);
    }); */
  });
  