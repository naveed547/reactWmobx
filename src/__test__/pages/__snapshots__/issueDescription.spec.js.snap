// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueDescription IssueDescription component renders correctly 1`] = `
<div>
  <h2
    className="my-5 text-center"
  >
    Issue Description
  </h2>
  User Name: 
  twbs
   
  <br />
  Repo Name: 
  bootstrap
   
  <br />
  <div
    className="card"
  >
    <div
      className="card-header row mx-0"
    >
      <div
        className="col-sm-10 pl-0"
      >
        #
        363302203
         
        <b>
          Add license headers in js/dist files
        </b>
      </div>
      <div
        className="col-sm-2 text-right text-success pr-0"
      >
        <i
          className="fa fa-check text-success"
        />
        <span
          className="pl-1 text-capitalize"
        >
          closed
        </span>
      </div>
    </div>
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        Issue Number: 
        <b>
          27301
        </b>
      </p>
      <p
        className="card-text"
      >
        
      </p>
    </div>
    <div
      className="card-body row"
    >
      <p
        className="card-text col-sm-6"
      >
        <span>
          Created By: 
        </span>
        <span>
          <b>
            XhmikosR
          </b>
        </span>
      </p>
      <p
        className="card-text col-sm-6 text-right"
      >
        <small
          className="text-muted"
        >
          Last updated 
          <b>
            9 days ago
          </b>
        </small>
      </p>
    </div>
  </div>
</div>
`;
